opal/corelib/range.rb

```diff
@@ -85,16 +93,26 @@ def exclude_end?
     @exclude
   end
 
-  alias :first :begin
+  def first(n=(n_specified=true; nil))
```

Comment:
`(n = undefined)` produces uninitialized function argument, it is `undefined` (from JS world) if you don't pass any value.

You can use `undefined` (from Ruby) only in optional arguments. By using it inline you will get a method call :smile: 
``` ruby
$ opal --compile --no-opal -e "a = undefined"

return (a = self.$undefined())

```